x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/application/EnterpriseSearchFeatureSetUsage.java

```diff
@@ -21,40 +21,55 @@
 
 public class EnterpriseSearchFeatureSetUsage extends XPackFeatureSet.Usage {
 
+    static final TransportVersion BEHAVIORAL_ANALYTICS_TRANSPORT_VERSION = TransportVersion.V_8_8_1;
+    static final TransportVersion QUERY_RULES_TRANSPORT_VERSION = TransportVersion.V_8_500_046;
+
     public static final String SEARCH_APPLICATIONS = "search_applications";
     public static final String ANALYTICS_COLLECTIONS = "analytics_collections";
+    public static final String QUERY_RULESETS = "query_rulesets";
     public static final String COUNT = "count";
     private final Map<String, Object> searchApplicationsUsage;
     private final Map<String, Object> analyticsCollectionsUsage;
+    private final Map<String, Object> queryRulesUsage;
 
     public EnterpriseSearchFeatureSetUsage(
         boolean available,
         boolean enabled,
         Map<String, Object> searchApplicationsUsage,
-        Map<String, Object> analyticsCollectionsUsage
+        Map<String, Object> analyticsCollectionsUsage,
+        Map<String, Object> queryRulesUsage
     ) {
         super(XPackField.ENTERPRISE_SEARCH, available, enabled);
         this.searchApplicationsUsage = Objects.requireNonNull(searchApplicationsUsage);
         this.analyticsCollectionsUsage = Objects.requireNonNull(analyticsCollectionsUsage);
+        this.queryRulesUsage = Objects.requireNonNull(queryRulesUsage);
     }
 
     public EnterpriseSearchFeatureSetUsage(StreamInput in) throws IOException {
         super(in);
         this.searchApplicationsUsage = in.readMap();
-        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_8_1)) {
+        if (in.getTransportVersion().onOrAfter(QUERY_RULES_TRANSPORT_VERSION)) {
+            this.analyticsCollectionsUsage = in.readMap();
+            this.queryRulesUsage = in.readMap();
+        } else if (in.getTransportVersion().onOrAfter(BEHAVIORAL_ANALYTICS_TRANSPORT_VERSION)) {
             this.analyticsCollectionsUsage = in.readMap();
+            this.queryRulesUsage = Collections.emptyMap();
         } else {
             this.analyticsCollectionsUsage = Collections.emptyMap();
+            this.queryRulesUsage = Collections.emptyMap();
```

Comment:
As a suggestion, we could clarify a bit with the following:

```suggestion
        Map<String, Object> analyticsCollectionsUsage = Collections.emptyMap();
        Map<String, Object> queryRulesUsage = Collections.emptyMap();

        if (in.getTransportVersion().onOrAfter(BEHAVIORAL_ANALYTICS_TRANSPORT_VERSION)) {
            analyticsCollectionsUsage = in.readMap();
        }
        if (in.getTransportVersion().onOrAfter(QUERY_RULES_TRANSPORT_VERSION)) {
            queryRulesUsage = in.readMap();
        } 
        this.analyticsCollectionsUsage = analyticsCollectionsUsage;
        this.queryRulesUsage = queryRulesUsage;
```